### ISSUE TITLE
[BUG] - No Date header in test email?

### ISSUE THREAD

**Issue by SwissOS** [mealie-recipes/mealie#3028]
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue (title field is above this).
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [X] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

I am trying to setup SMTP service via my local email server. I use Postfix with OpenDMARC and OpenDKIM.
When mealie tries to send a test email, OpenDMARC complains with this error:
``RFC5322 requirement error: not exactly one Date field``
I was wondering, how to check if Mealie added a date header to the email?

[From RFC5322](https://datatracker.ietf.org/doc/html/rfc5322#section-3.6):
"The only required header fields are the origination date field and the originator address field(s).  All other header fields are
   syntactically optional."

### Steps to Reproduce

In Site Settings, under the Email section, try to send an email with "Test" button

### Please provide relevant logs

ERROR: 22-Jan-24 12:42:18       (550, b'5.7.1 Command rejected')

### Mealie Version

Docker image 1.0 release

### Deployment

Docker (Linux)

**Comment by Kuchenpirat** [comment]
I have mealie-email configured with my gmail workspace. 
When opening the mail header, it shows a date field. As far as i know, the library (smtplib) we use automatically adds the Date header in the Mail. 

As i do not have much experience with selfhosting email, maybe you could temporarily disable the rule that enforces RFC5322 compliancy and check the header afterwards.

**Comment by Kuchenpirat** [comment]
Return-Path: <sender>
Received: from [192.168.0.1] (ip-000-000-000-000.um00.pools.isp-ip.de. [0.0.0.0])
        by smtp.email.com with ESMTPSA id l0-00000a0d0000000000b0000000000000sm0000000wrv.00.0000.00.00.00.00.00
        for <reciever>
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Mon, 22 Jan 2024 07:53:59 -0800 (PST)
Message-ID: <00000000.00000000.0000.0000@mx.google.com>
Date: Mon, 22 Jan 2024 07:53:59 -0800 (PST)
Subject: Test Email
From: App <sender>
To: reciever
Content-Type: multipart/alternative; boundary="===============0000000000000000000=="

This is the header i can see in my gmail under send (heavily anonymized)
Gmail then adds a bunch of DKIM stuff and signatures.

**Comment by SwissOS** [comment]
Thanks @Kuchenpirat for the suggestion. I have modified my openDMARC settings to remove header verification, and I did receive the test email.
But just to be clear, the test email does not add the date header, now I am 100% sure. Here's the output of my spam filter:
```
Content preview:  Test Email Hi there! This is a test email. 

Content analysis details:   (5.4 points, 5.0 required)

 pts rule name              description
---- ---------------------- --------------------------------------------------
 1.4 MISSING_DATE           Missing Date: header
 0.1 MISSING_MID            Missing Message-Id: header
```

In your case, the date you're seeing in Gmail is simply the date Gmail received the email, not the date header of the email; most likely the Gmail SMTP server added the header.

Thanks for letting me know about smtplib. I see that this problem was already discussed way back in ... 2016! See [this page for that discussion](https://bugs.python.org/issue28879).

But it seems that in the end, it's up to the programmer to add his headers, and the suggestion for smtplib to automatically add the date header was never implemented. See this issue here: [https://github.com/python/cpython/issues/107809](https://github.com/python/cpython/issues/107809)

All in all, I would suggest you add the Date header in Mealie. It's probably only a minor issue (something like: ``msg["Date"] = email.utils.formatdate(localtime=True)`` ?) as discussed in the second linked issue.
